### PR TITLE
Sprint 8

### PR DESCRIPTION
**Pull request by RastelliBru**

### CODE REVIEW

File: account/repositories/accountRepository.js [hunk 1]
```diff
@@ -1,15 +1,22 @@
 import { MongoClient } from "mongodb";
+import * as dotenv from 'dotenv'
```
Comment (DPechetti) [inline account/repositories/accountRepository.js:2]: the [documentation](https://github.com/motdotla/dotenv#usage) says: As early as possible in your application, import and configure dotenv

but i don't think you did

File: account/repositories/accountRepository.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+export async function getUsersCollection(client) {
+  const database = client.db("accounts");
+  const usersCollection = database.collection('users')
+  return usersCollection;
 }
 
 export async function saveAccount(account) {
-  const collection = await getUsersCollection();
-  await collection.insertOne(account);
+  await client.connect();
```
Comment (DPechetti) [inline account/repositories/accountRepository.js:15]: will you create a connection everytime you save an account?

File: account/repositories/accountRepository.js [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
+  const usersCollection = await getUsersCollection(client);
+  await usersCollection.insertOne(account);
+  setTimeout(() => {client.close()}, 1500)
+  await client.close();
```
Comment (DPechetti) [inline account/repositories/accountRepository.js]: why do you have this implementation in your save function?

File: account/src/main.js [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
-const app = express();
+import { app } from "./app.js";
 
 app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(apiDocs))
 
-app.use(express.json());
-app.use(router);
-
-app.use("/accountsV1", router)
+//app.use("/accountsV1", router)
```
Comment (DPechetti) [inline account/src/main.js]: i don't think you need this comment

File: account/src/routes.js [hunk 5]
```diff
@@ -2,14 +2,15 @@ import { Router } from "express";
...
   const { name, email, password } = req.body;
   const encodedPassword = bcrypt.hashSync(password, 10);
-  createUserUseCase(name, email, encodedPassword)
+  const user = await createUserUseCase(name, email, encodedPassword)
     .then((user) => {
       const userCreated = {
+        id: user.id,
         name: user.name,
         email: user.email,
         date: user.createdDate,
```
Comment (DPechetti) [inline account/src/routes.js:16]: isn't a good practice to change the data received from use case

File: account/src/routes.js [hunk 6]
```diff
@@ -2,14 +2,15 @@ import { Router } from "express";
...
-export const router = Router();
+const router = Router();
 
-router.post("/account", function (req, res) {
+router.post("/account", async (req, res) => {
   const { name, email, password } = req.body;
   const encodedPassword = bcrypt.hashSync(password, 10);
-  createUserUseCase(name, email, encodedPassword)
+  const user = await createUserUseCase(name, email, encodedPassword)
     .then((user) => {
```
Comment (DPechetti) [inline account/src/routes.js:11]: why are you using await and then at same function?

File: account/src/routes.js (hunk 5 above)
Comment (DPechetti) [inline account/src/routes.js:16]: can you tell me what this variable does?

File: account/test/createUserAccount.test.js [hunk 7]
```diff
@@ -1,5 +1,31 @@
...
+import request from 'supertest';
+import { client, getUsersCollection } from '../repositories/accountRepository.js';
+import { app } from '../src/app.js';
 
-const user = createUserUseCase("Bruna", "bruna@dominio.com", "123senhA#");
-
-console.log(user);
+describe('Account Creation', () => {
+    afterEach( async() => {
+        await client.connect();
```
Comment (DPechetti) [inline account/test/createUserAccount.test.js:7]: are you creating another connection?

File: product/src/repositories/connectiondb.js [hunk 8]
```diff
@@ -1,5 +1,9 @@
 import { Sequelize } from 'sequelize'
+import * as dotenv from 'dotenv'
```
Comment (DPechetti) [inline product/src/repositories/connectiondb.js:2]: same here

File: product/src/repositories/products.js [hunk 9]
```diff
@@ -1,19 +1,22 @@
+import { randomUUID } from 'crypto'
```
Comment (DPechetti) [inline product/src/repositories/products.js:1]: where is your database?

**Review by DPechetti — COMMENTED** [review]
I think you created a package.lock in the root by mistake

**Review by DPechetti — APPROVED** [review]

File: account/repositories/accountRepository.js [hunk 10]
```diff
@@ -1,15 +1,21 @@
 import { MongoClient } from "mongodb";
+import * as dotenv from 'dotenv'
```
Comment (wandersonmaceds) [inline account/repositories/accountRepository.js:2]: @RastelliBru como você instalou o env-cmd, você pode remover essa linha e desinstalar o `dotenv`

File: product/_test_/e2e/createProduct.test.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+                expect(body).toEqual({
+                    ...produto1,
+                    createdDate: expect.any(String),
+                    productId: expect.any(String),
+                    features: produto1.features.map(feature => ({
+                        ...feature
+                    })),
+                    images: produto1.images.map(image => ({
+                        ...image 
+                    }))
```
Comment (wandersonmaceds) [inline product/_test_/e2e/createProduct.test.js:29]: esse teste passou sem precisar usar o arrayContaining?

File: product/_test_/e2e/listProducts.test.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+                createdAt: expect.any(String),
+                updatedAt: expect.any(String),
+                features: produto1.features.map(feature => ({
+                    ...feature,
+                    createdAt: expect.any(String),
+                    product_id: body[0].id,
+                    updatedAt: expect.any(String),
+                    id: expect.any(Number),
+                })),
+                images: produto1.images.map(image => ({
```
Comment (wandersonmaceds) [inline product/_test_/e2e/listProducts.test.js:39]: aqui não precisou do arrayContaining?

File: product/src/repositories/products.js [hunk 13]
```diff
@@ -1,19 +1,22 @@
...
+
 export const produto1 = {
-  nome: "the Sims 3",
-  valor: "99",
-  quantidade: 1,
-  descricao: "description",
-  categoria: "games",
-  caracteristicas: [
+  id_user: randomUUID(),
```
Comment (wandersonmaceds) [inline product/src/repositories/products.js:4]: esse arquivo não deveria estar nessa pasta.

**Pull request merged**